sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudJobIntegrationTests.cs

```diff
@@ -939,7 +939,60 @@ public async Task Job_GetTaskCounts_ReturnsCorrectCount()
 
                         var counts = await unboundJob.GetTaskCountsAsync().ConfigureAwait(false);
 
-                        Assert.Equal(2, counts.Active);
+                        Assert.Equal(2, counts.TaskCounts.Active);
+                        Assert.Equal(2, counts.TaskSlotCounts.Active);
+                    }
+                    finally
+                    {
+                        await TestUtilities.DeleteJobIfExistsAsync(batchCli, jobId);
+                    }
+                }
+            };
+
+            await SynchronizationContextHelper.RunTestAsync(test, TestTimeout);
+        }
+
+
+        [Fact]
+        [LiveTest]
+        [Trait(TestTraits.Duration.TraitName, TestTraits.Duration.Values.ShortDuration)]
+        public async Task Job_GetTaskCounts_ReturnsCorrectCountNonZeroTaskSlots()
+        {
+            Func<Task> test = async () =>
+            {
+                using (BatchClient batchCli = await TestUtilities.OpenBatchClientFromEnvironmentAsync().ConfigureAwait(false))
+                {
+                    string jobId = "NonZeroTaskSlots-" + TestUtilities.GetMyName();
+                    try
+                    {
+                        PoolInformation poolInfo = new PoolInformation()
+                        {
+                            PoolId = "Fake"
+                        };
+
+                        CloudJob unboundJob = batchCli.JobOperations.CreateJob(jobId, poolInfo);
+                        await unboundJob.CommitAsync().ConfigureAwait(false);
+                        await unboundJob.RefreshAsync().ConfigureAwait(false);
+
+                        CloudTask t1 = new CloudTask("t1", "cmd /c dir");
+                        t1.RequiredSlots = 2;
+                        CloudTask t2 = new CloudTask("t2", "cmd /c ping 127.0.0.1 -n 4");
+                        t2.RequiredSlots = 3;
+
+                        await unboundJob.AddTaskAsync(new[] { t1, t2 }).ConfigureAwait(false);
+
+                        await Task.Delay(TimeSpan.FromSeconds(5)).ConfigureAwait(false); // Give the service some time to get the counts
```

Comment:
Relying on time might lead to a flaky test. Is there something we can poll or wait on, instead?

Reply:
Unfortunately not. This part of the service is actually itself flaky so our best time is time wait

Reply:
Let's discuss this on a call.